ci/src/mainnet_revisions/mainnet_revisions.py

```diff
@@ -0,0 +1,177 @@
+    logging.info("Current NNS subnet (%s) revision: %s", nns_subnet_id, current_nns_version)
+    current_app_subnet_version = get_subnet_replica_version(app_subnet_id)
+    logging.info("Current App subnet (%s) revision: %s", app_subnet_id, current_app_subnet_version)
+
+    ic_repo_push_token = os.environ.get("PUSH_TO_IC", "ENV_VAR_NOT_SET")
+    repo = "dfinity/ic"
+
+    repo_root = pathlib.Path(os.environ.get("BUILD_WORKING_DIRECTORY", "NOT_PRESENT"))
+
+    if not repo_root.parent.exists():
+        raise Exception ("Expected dir %s to exist", repo_root.name)
+
+    running_on_ci = os.environ.get("CI") or os.environ.get("GITHUB_ACTION")
+
+    branch = "ic-mainnet-revisions"
+    subprocess.call(["git", "fetch"], cwd=repo_root)
+    if subprocess.call(["git", "checkout", branch], cwd=repo_root) == 0:
+        # The branch already exists, update the existing MR
+        logging.info("Found an already existing target branch")
+    else:
+        subprocess.check_call(["git", "checkout", "-b", branch], cwd=repo_root)
+    subprocess.check_call(["git", "reset", "--hard", "origin/master"], cwd=repo_root)
+
+    update_saved_subnet_version(subnet=nns_subnet_id, version=current_nns_version, repo_root=pathlib.Path(repo_root))
+    update_saved_subnet_version(
+        subnet=app_subnet_id, version=current_app_subnet_version, repo_root=pathlib.Path(repo_root)
+    )
+    git_modified_files = subprocess.check_output(["git", "ls-files", "--modified", "--others"], cwd=repo_root).decode(
+        "utf8"
+    )
+    if SAVED_VERSIONS_PATH in git_modified_files:
+        logging.info("Creating/updating a MR that updates the saved NNS subnet revision")
+        subprocess.check_call(["git", "add", SAVED_VERSIONS_PATH], cwd=repo_root)
+        if running_on_ci:
+            subprocess.check_call(["git", "config", "user.email", "infra@dfinity.org"], cwd=repo_root)
+            subprocess.check_call(["git", "config", "user.name", "CI Automation"], cwd=repo_root)
+        subprocess.check_call(
+            ["git", "commit", "-m", "chore: Update Mainnet IC revisions file", SAVED_VERSIONS_PATH],
+            cwd=repo_root,
+        )
+        subprocess.check_call(
+            ["git", "push", "origin", branch, "-f"],
+            cwd=repo_root
+        )
+        gh_env = os.environ.copy()
+        gh_env["GITHUB_TOKEN"] = ic_repo_push_token
+
+        if not subprocess.check_output(
+            ["gh", "pr", "list", "--head", branch, "--repo", repo],
+            cwd=repo_root,
+            env=gh_env,
+        ).decode("utf8"):
+            subprocess.check_call(
+                [
+                    "gh",
+                    "pr",
+                    "create",
+                    "--head",
+                    branch,
+                    "--repo",
+                    repo,
+                    "--fill",
+                ],
+                cwd=repo_root,
+                env=gh_env,
+            )
```

Comment:
Note that the default behavior of subprocess is to inhert the current process’ environment so I think you can just remove setting `env`.